### PR TITLE
Add new schema to save generated target of event marker

### PR DESCRIPTION
**Pull request by shueja**
Also fix logic for stale trajectory marker.

Related to #401, should fix the event markers being zeroed out since we no longer rely on isTrajectoryStale to deserialize them.

### CODE REVIEW

File: src/document/HolonomicPathStore.ts [hunk 1]
```diff
@@ -807,17 +798,19 @@ export const HolonomicPathStore = types
...
-          return true;
+          // Reaction needs the return value to change,
+          // so we can't just access the values and do nothing with them
+          
+          return {
+            waypoints: toJS(self.waypoints),
+            constraints: toJS(self.constraints),
+            // does not need toJS to do a deep check on this, since it's just a boolean
+            guessing: self.usesControlIntervalGuessing,
+            obstacles: toJS(self.obstacles)
```
Comment (Awesomeplayer165) [inline src/document/HolonomicPathStore.ts:810]: should this be a type?

File: src/document/previousSpecs/v0_3_1.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+export const SAVE_FILE_VERSION = "v0.3.1";
+export interface SavedWaypoint {
+  x: number;
+  y: number;
+  heading: number;
+  isInitialGuess: boolean;
+  translationConstrained: boolean;
+  headingConstrained: boolean;
+  controlIntervalCount: number; // positive-integer>
```
Comment (Awesomeplayer165) [inline src/document/previousSpecs/v0_3_1.ts:13]: make comments more uniform
```suggestion
  controlIntervalCount: number; // positive-integer
```

File: src/document/previousSpecs/v0_3_1.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+}
+export type SavedGeneratedWaypoint = SavedWaypoint & {
+  timestamp: number;
+  isStopPoint: boolean;
+};
+export interface SavedTrajectorySample {
+  timestamp: number; //positive
```
Comment (Awesomeplayer165) [inline src/document/previousSpecs/v0_3_1.ts:20]:
```suggestion
  timestamp: number; // positive
```

**Review by Awesomeplayer165 — CHANGES_REQUESTED** [review]
looks good. keep going!

File: src/document/HolonomicPathStore.ts (hunk 1 above)
Comment (shueja) [reply src/document/HolonomicPathStore.ts:810]: No, it's not needed since this structure only exists in one spot.

**Review by Awesomeplayer165 — APPROVED** [review]

**Pull request merged**